vent/coordinator/coordinator.py

```diff
@@ -72,127 +62,146 @@ def __init__(self, sim_mode=False):
-        # TODO: pass max_heartbeat_interval
-        self.process_manager = ProcessManager()
-        self.rpc = IPC()
+        # TODO: according to documentation, pass max_heartbeat_interval?
+        self.process_manager = ProcessManager(sim_mode)
+        self.proxy = get_rpc_client()
+        # TODO: make sure the ipc connection is setup. There should be a clever method
+
+    def get_sensors(self) -> Dict[ValueName, SensorValueNew]:
+        sensor_values = pickle.loads(self.proxy.get_sensors().data)
+        res = {
+            ValueName.PIP: SensorValueNew(ValueName.PIP, sensor_values.pip, sensor_values.timestamp,
+                                          sensor_values.loop_counter),
+            ValueName.PEEP: SensorValueNew(ValueName.PEEP, sensor_values.peep, sensor_values.timestamp,
+                                           sensor_values.loop_counter),
+            ValueName.FIO2: SensorValueNew(ValueName.FIO2, sensor_values.fio2, sensor_values.timestamp,
+                                           sensor_values.loop_counter),
+            ValueName.TEMP: SensorValueNew(ValueName.TEMP, sensor_values.temp, sensor_values.timestamp,
+                                           sensor_values.loop_counter),
+            ValueName.HUMIDITY: SensorValueNew(ValueName.HUMIDITY, sensor_values.humidity, sensor_values.timestamp,
+                                               sensor_values.loop_counter),
+            ValueName.PRESSURE: SensorValueNew(ValueName.PRESSURE, sensor_values.pressure, sensor_values.timestamp,
+                                               sensor_values.loop_counter),
+            ValueName.VTE: SensorValueNew(ValueName.VTE, sensor_values.vte, sensor_values.timestamp,
+                                          sensor_values.loop_counter),
+            ValueName.BREATHS_PER_MINUTE: SensorValueNew(ValueName.BREATHS_PER_MINUTE, sensor_values.breaths_per_minute,
+                                                         sensor_values.timestamp, sensor_values.loop_counter),
+            ValueName.INSPIRATION_TIME_SEC: SensorValueNew(ValueName.INSPIRATION_TIME_SEC,
+                                                           sensor_values.inspiration_time_sec, sensor_values.timestamp,
+                                                           sensor_values.loop_counter),
+        }
+        return res
+
+    def get_active_alarms(self) -> Dict[str, Alarm]:
+        return pickle.loads(self.proxy.get_active_alarms())
+
+    def get_logged_alarms(self) -> List[Alarm]:
+        return pickle.loads(self.proxy.get_logged_alarms())
+
+    def clear_logged_alarms(self):
+        # TODO: implement this
         raise NotImplementedError
 
+    def set_control(self, control_setting: ControlSetting):
+        self.proxy.set_control(pickle.dumps(control_setting))
+
+    def get_control(self, control_setting_name: ValueName) -> ControlSetting:
+        return pickle.loads(self.proxy.get_control(pickle.dumps(control_setting_name)).data)
```

Comment:
Also here, I'd rather have no nesting of calls on one line, for ease of reading and future debugging.